save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/service/AgentService.kt

```diff
@@ -119,7 +119,25 @@ class AgentService(
                     getFinishedOrStoppedAgentsByExecutionId(executionId)
                 )
             }
-            .filter { finishedContainerIds -> finishedContainerIds.isNotEmpty() }
+            .filter { finishedContainerIds ->
+                finishedContainerIds.isNotEmpty()
+                    .also { hasFinishedContainers ->
+                        if (!hasFinishedContainers) {
+                            log.debug { "Agents for execution $executionId are still running, so won't try to stop them" }
+                        }
+                    }
+            }
+            .filterWhen {
+                orchestratorAgentService.getExecutionStatus(executionId)
+                    .map {
+                        it !in setOf(ExecutionStatus.FINISHED, ExecutionStatus.FINISHED)
```

Comment:
set of `finished`, `finished`?

Reply:
good catch, thanks. Should be `FINISHED` and `ERROR`